metpy/calc/tools.py

```diff
@@ -2,17 +2,24 @@
 # Distributed under the terms of the BSD 3-Clause License.
 # SPDX-License-Identifier: BSD-3-Clause
 """Contains a collection of generally useful calculation tools."""
+from __future__ import division
 
 import functools
 import warnings
 
 import numpy as np
+try:
+    from numpy.core.numeric import normalize_axis_index
```

Comment:
Should really be `np.core.multiarray` instead

Reply:
That makes more sense, good to know. I'll look at a PR to numpy some time when I have some cycles...

Not sure what I'd call the damned thing 😁

Reply:
Refactoring `np.gradient` to use your `_first_derivative` would be a good first step

Reply:
There's a pretty steep performance penalty (or it would seem from the complexity) for the support for irregular spacing though...

Reply:
We already have irregular spacing - all we don't currently support is broadcasting of dx against the values.